client/graphite/write.go

```diff
@@ -73,17 +74,22 @@ func (c *Client) disconnectFromCarbon() {
 }
 
 // Write implements the client.Writer interface.
-func (c *Client) Write(samples model.Samples) error {
+func (c *Client) Write(samples model.Samples, urlValues url.Values) error {
 	if c.cfg.Write.CarbonAddress == "" {
 		return nil
 	}
 
+	graphitePrefix := c.cfg.DefaultPrefix
+	if urlValues.Get("graphite.default-prefix") != "" {
```

Comment:
how do we make the difference between an empty prefix and a non-specified one ?

Reply:
If there are no values associated with the key, Get returns the empty string.
If there is no custom path then it will be empty string too.

Reply:
Values is a map[string]string, you can use the map directly to check if a key exists or not.
https://stackoverflow.com/questions/2050391/how-to-check-if-a-map-contains-a-key-in-go

Reply:
I'm using the method described in the url.go:
```
type Values map[string][]string

// Get gets the first value associated with the given key.
// If there are no values associated with the key, Get returns
// the empty string. To access multiple values, use the map
// directly.
func (v Values) Get(key string) string {
	if v == nil {
		return ""
	}
	vs := v[key]
	if len(vs) == 0 {
		return ""
	}
	return vs[0]
}```

Reply:
That's my point, don't use it. We want to be able to specify an empty prefix using the parameters.